wallet/src/de/schildbach/wallet/adapter/BuyAndSellDashServicesAdapter.kt

```diff
@@ -17,82 +17,128 @@
 
 package de.schildbach.wallet.adapter
 
-
-import android.annotation.SuppressLint
 import android.view.LayoutInflater
 import android.view.ViewGroup
 import androidx.core.content.ContextCompat
 import androidx.core.view.isGone
 import androidx.core.view.isVisible
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.RecyclerView
 import de.schildbach.wallet.data.BuyAndSellDashServicesModel
+import de.schildbach.wallet.data.ServiceStatus
 import de.schildbach.wallet_test.databinding.ItemServiceListBinding
 import org.bitcoinj.core.Coin
-import org.dash.wallet.common.Configuration
 import org.dash.wallet.common.Constants
-import org.dash.wallet.common.ui.BaseAdapter
 import org.dash.wallet.common.util.GenericUtils
+import androidx.recyclerview.widget.ListAdapter
+import de.schildbach.wallet_test.R
+import org.bitcoinj.utils.MonetaryFormat
+
+
+class BuyAndSellDashServicesAdapter(
+    val balanceFormat: MonetaryFormat,
+    val onClickListener: (BuyAndSellDashServicesModel) -> Unit
+) : ListAdapter<BuyAndSellDashServicesModel, BuyAndSellDashServicesAdapter.BuyAndSellDashServicesViewHolder>(
+    BuyAndSellDashDiffCallback()
+) {
+    class BuyAndSellDashDiffCallback : DiffUtil.ItemCallback<BuyAndSellDashServicesModel>() {
+        override fun areItemsTheSame(oldItem: BuyAndSellDashServicesModel, newItem: BuyAndSellDashServicesModel): Boolean {
+            return oldItem.serviceType == oldItem.serviceType
+        }
+
+        override fun areContentsTheSame(oldItem: BuyAndSellDashServicesModel, newItem: BuyAndSellDashServicesModel): Boolean {
+            return oldItem == newItem
+        }
+    }
 
-class BuyAndSellDashServicesAdapter( val config: Configuration,
-                                     val onClickListener: (BuyAndSellDashServicesModel) -> Unit) : BaseAdapter<BuyAndSellDashServicesModel>(){
-    var isOnline: Boolean = false
-    override fun viewHolder(layout: Int, view: ViewGroup): BaseViewHolder {
-        return BuyAndSellDashServicesViewHolder(
-            ItemServiceListBinding.inflate(LayoutInflater.from(view.context), view, false)
-        )
+    override fun onCreateViewHolder(
+        parent: ViewGroup,
+        viewType: Int
+    ): BuyAndSellDashServicesViewHolder {
+        val inflater = LayoutInflater.from(parent.context)
+        val binding = ItemServiceListBinding.inflate(inflater, parent, false)
 
+        return BuyAndSellDashServicesViewHolder(binding)
     }
 
-    fun updateIconState(hasInternet: Boolean) {
-        isOnline = hasInternet
-        notifyDataSetChanged()
```

Comment:
The items already have a field `serviceStatus` that is assigned, among other things, based on network availability. We can use it for graying out icons as well - it will allow getting rid of `notifyDataSetChanged` which is bad for performance.